### ISSUE TITLE
`git_commit` action doesn't escape its `message` option

### ISSUE THREAD

**Issue by rpowelll** [fastlane/fastlane#4445]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

The `git_commit` command passes any text passed via its `message` option to the underlying `git commit` command via the `-m` command line flag, but neglects to escape characters that have special significance when used in the context of a command shell. This means that if a user includes characters such as brackets or quotation marks in the message in question, the action will fail.

##### Complete output when running fastlane, including the stack trace and command used

```
[15:15:39]: ------------------------
[15:15:39]: --- Step: git_commit ---
[15:15:39]: ------------------------
[15:15:39]: $ git commit -m 'Released v0.1.0-build.15 :rocket:

- NEW: Browse a contact's documents' .
[15:15:39]: ▸ error: pathspec 'documents' did not match any file(s) known to git.
[15:15:39]: ▸ sh: line 3: -: command not found
[15:15:39]: ▸ sh: -c: line 4: unexpected EOF while looking for matching `''
[15:15:39]: ▸ sh: -c: line 5: syntax error: unexpected end of file
[15:15:39]: Variable Dump:
[15:15:39]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :VERSION_NUMBER=>"0.1.0", :BUILD_NUMBER=>"15", :SIGH_PROFILE_TYPE=>"app-store", :IPA_OUTPUT_PATH=>"/Users/rpowell/Code/DomainAgentApp/REDACTED.ipa", :DSYM_OUTPUT_PATH=>"/Users/rpowell/Code/DomainAgentApp/REDACTED.app.dSYM.zip", :XCODEBUILD_ARCHIVE=>"/Users/rpowell/Library/Developer/Xcode/Archives/2016-05-02/REDACTED 2016-05-02 14.58.41.xcarchive"}
[15:15:39]: Exit status of command 'git commit -m 'Released v0.1.0-build.15 :rocket:

- NEW: Browse a contact's documents' .' was 2 instead of 0.
error: pathspec 'documents' did not match any file(s) known to git.
sh: line 3: -: command not found
sh: -c: line 4: unexpected EOF while looking for matching `''
sh: -c: line 5: syntax error: unexpected end of file
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.70.0"

default_platform :ios

platform :ios do
  before_all do |lane, options|
    ENV["SLACK_URL"] = "https://hooks.slack.com/services/TOKEN"

    # Make this work a little better on CI

    unless options[:skip_dependencies]
      # Only install Carthage dependencies if they haven't changed
      unless check_files_identical(files: ["Cartfile.resolved", "Carthage/Cartfile.resolved"])
        carthage(platform: "iOS", use_binaries: false)
        sh "cp Cartfile.resolved Carthage/Cartfile.resolved"
      end

      download_fabric_frameworks(
        fabric: "1.6.7",
        crashlytics: "3.7.0"
      )
    end
  end

  desc "Install development dependencies"
  lane :dependencies do
    # Skip certificates on CI for now
    unless ENV["CIRCLECI"]
      match(type: "development")
      match(type: "appstore")
    end
  end

  desc "Produce a signed IPA"
  lane :ipa do
    match(type: "appstore")
    gym(scheme: "REDACTED", output_directory: "./build", configuration: "debug")
  end

  desc "Runs all the tests"
  lane :test do
    output_directory = ENV["SCAN_OUTPUT_DIRECTORY"] || "./fastlane/test_output"
    scan(scheme: "REDACTED", output_directory: "#{output_directory}/REDACTED")
    scan(scheme: "REDACTED-iOS", output_directory: "#{output_directory}/REDACTED-iOS")
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    version = get_version_number
    build = increment_build_number
    semantic_version = "#{version}-build.#{build}"
    changelog = prompt(text: "Enter the change log: ", multi_line_end_keyword: "END")

    # Deploy new build
    match(type: "appstore") # more information: https://codesigning.guide
    gym(scheme: "REDACTED") # Build your app - more options available
    pilot(changelog: changelog)
    slack(
      channel: "agent-notifications",
      message: "Successfully distributed TestFlight build #{build} :rocket:\n\n#{changelog}"
    )

    # Commit changes back to git
    git_commit(path: ".", message: "Released v#{semantic_version} :rocket:\n\n#{changelog}")
    add_git_tag(tag: semantic_version)
    push_to_git_remote
    set_github_release(
      repository_name: "domain-group/agent-app",
      name: "v#{version} (#{build})",
      tag_name: semantic_version,
      description: changelog,
      is_draft: false,
      upload_assets: ["REDACTED.app.dSYM.zip", "REDACTED.ipa"]
    )
  end

  desc "Deploy a new version to the App Store"
  lane :appstore do
    increment_build_number
    match(type: "appstore")
    # snapshot
    gym(scheme: "REDACTED") # Build your app - more options available
    deliver(force: true)
    # frameit
    slack(
      channel: "agent-notifications",
      message: "Successfully distributed App Store build #{version} :rocket:"
    )
  end
end
```

### Environment

fastlane version (run `fastlane -v`):
1.83.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?
yes

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?
yes

**Comment by mfurtak** [comment]
Hi @rpowelll - thanks for the report! 👍 

As you are finding, it is a mistake for these actions to wrap their command parameters in single quotes. For example:

```ruby
Actions.sh("git tag -am '#{message}' '#{tag}'")
```

Instead, these command parameters should be escaped using the `shellwords` module's `shellescape` method _without_ trying to wrap the values in quotes. For example:

```ruby
# At top of file...
require 'shellwords'

# Later on...
Actions.sh("git tag -am #{message.shellescape} #{tag.shellescape}")
```

Do you think you can adjust your fix in #4446 to try this approach?

**Comment by mfurtak** [comment]
Accepted as #4446 - thanks! 🚀
